From dd14a1b17a154d15f6364709c592b7af235511b7 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 5 Jul 2024 09:05:36 +0200
Subject: [PATCH 1184/1695] plat-stm32mp1: don't register scmi server PM if not
 needed

Do not embed SCMI server PM callbacks when not needed so that they
don't waste unpaged memory space when pager is disabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f5e95ba29d54f0eba030f2c43c09d24314b91ab1 plat-stm32mp1: don't register scmi server PM if not needed]
Change-Id: I3144fc56f4fd8268ef02c392fb7fd8039c636224
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/392002
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/scmi_server.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/scmi_server.c b/core/arch/arm/plat-stm32mp1/scmi_server.c
index 278e98e0d..f680cc7ae 100644
--- a/core/arch/arm/plat-stm32mp1/scmi_server.c
+++ b/core/arch/arm/plat-stm32mp1/scmi_server.c
@@ -206,7 +206,7 @@ struct stm32_scmi_perfd scmi_performance_domain[] = {
 static const struct channel_resources scmi_channel[] = {
 	[0] = {
 		.channel = &(struct scmi_msg_channel){
-#ifdef SMT_BUFFER_BASE
+#if CFG_STM32MP1_SCMI_SHM_BASE
 			.shm_addr = { .pa = SMT_BUFFER_BASE },
 			.shm_size = SMT_BUF_SLOT_SIZE,
 #endif
@@ -780,7 +780,10 @@ static TEE_Result stm32mp1_init_scmi_server(void)
 	size_t i = 0;
 	size_t j = 0;
 
-	register_pm_driver_cb(stm32_scmi_pm, NULL, "scmi-server-driver");
+	/* Only need to reset the SYSRAM shared memory on PM resume */
+	if (IS_ENABLED(CFG_STM32MP1_SCMI_SHM_BASE))
+		register_pm_driver_cb(stm32_scmi_pm, NULL,
+				      "scmi-server-driver");
 
 	for (i = 0; i < ARRAY_SIZE(scmi_channel); i++) {
 		const struct channel_resources *res = scmi_channel + i;
-- 
2.47.2

